src/main/java/com/thebois/models/world/World.java

```diff
@@ -179,6 +163,50 @@ private void postObstacleEvent(final int posX, final int posY) {
         Pawntastic.BUS.post(obstacleEvent);
     }
 
+    @Override
+    public Optional<IStructure> findNearestStructure(final Position position) {
+        final int searchRow = Math.round(position.getPosY());
+        final int searchCol = Math.round(position.getPosX());
+
+        final int maxSearchRadius = this.worldSize / 2;
```

Comment:
Having half the worldsize will lead to scenarios where they are at the edges and cant find anything right? Seems like that would cause a lot of frustration for the player. (Instead of reopen so it gets into the new review)

Reply:
![bild](https://user-images.githubusercontent.com/47867973/137345004-5479768a-1996-4268-be8c-0f13159ce8b9.png)
Like this